eurocropsml/dataset/utils.py

```diff
@@ -26,29 +26,32 @@
 logger = logging.getLogger(__name__)
 
 
-def read_files(data_dir: Path) -> list[str]:
+def read_files(data_dir: Path) -> set[str]:
     """Method to read all `.npy` files in a directory.
 
     Args:
         data_dir: Path that contains `.npy` files where labels and data are stored.
 
     Returns:
-        List of all `.npy` files that lie within the directory.
+        List of all `.npz` files that lie within the directory.
     """
-    return [file.name for file in data_dir.iterdir() if file.suffix == ".npz"]
+    return {file.name for file in data_dir.iterdir() if file.suffix == ".npz"}
 
 
 def _split_dataset(
     data_dir: Path,
+    satellite: list[Literal["S1", "S2"]],
     pretrain_classes: set[int],
     finetune_classes: set[int] | None = None,
     regions: set[str] | None = None,
 ) -> tuple[dict[int, list[str]], dict[int, list[str]] | None]:
-    files_list = read_files(data_dir)
+    files_list: set = set()
+    for s in satellite:
+        files_list.update(read_files(data_dir.joinpath(s)))
 
     # pre-filter by regions for region and regionclass split
     if regions is not None:
-        files_list = [file for file in files_list if file.startswith(tuple(regions))]
+        files_list = {file for file in files_list if file.startswith(tuple(regions))}
```

Comment:
Maybe change the variable naming to reflect that this is not a "list" but a set?

Reply:
Also done